### ISSUE TITLE
t/plugin-yaml.t sometimes fails

### ISSUE THREAD

**Issue by eserte** [jhthorsen/swagger2#95]
Sometimes running t/plugin-yaml.t fails like this:
```
$ perl5.22.2 -Mblib t/plugin-yaml.t
# YAML::XS 0.62
# Swagger2: Invalid spec:
# /paths/~1pets~1{petId}/get/responses/200: All of the oneOf rules match.
# /paths/~1pets~1{petId}/get/responses/default: All of the oneOf rules match.
# /paths/~1pets/post/responses/default: All of the oneOf rules match.
# /paths/~1pets/post/responses/201: All of the oneOf rules match.
# /paths/~1pets/get/responses/200: All of the oneOf rules match.
# /paths/~1pets/get/responses/default: All of the oneOf rules match. at /tmpfs/.cpan-build-cpansand/2016080806/Swagger2-0.84-0/blib/lib/Mojolicious/Plugin/Swagger2.pm line 80, <DATA> line 565.
not ok 1 - Could not load Swagger2 plugin using YAML::XS
#   Failed test 'Could not load Swagger2 plugin using YAML::XS'
#   at t/plugin-yaml.t line 30.
# YAML::Syck 1.29
ok 2 - GET /v1/pets
ok 3 - 200 OK
ok 4 - exact match for JSON Pointer "/0/id"
ok 5 - exact match for JSON Pointer "/0/name"
1..5
# Looks like you failed 1 test of 5.
```
I'll check if I found a reason for this (e.g. hash randomization).

**Comment by eserte** [comment]
Yes, it seems to be related to hash randomization. First find a `PERL_HASH_SEED` value for which the test fails:
```
$ for i in $(seq 100); do echo "hash seed: $i"; env PERL_HASH_SEED=$i PERL_PERTURB_KEYS=NO perl5.24.0 -Mblib t/plugin-yaml.t || break; done
...
hash seed: 25
# YAML::XS 0.62
# Swagger2: Invalid spec:
# /paths/~1pets/post/responses/default: All of the oneOf rules match.
# /paths/~1pets/post/responses/201: All of the oneOf rules match.
# /paths/~1pets/get/responses/200: All of the oneOf rules match.
# /paths/~1pets/get/responses/default: All of the oneOf rules match.
# /paths/~1pets~1{petId}/get/responses/200: All of the oneOf rules match.
# /paths/~1pets~1{petId}/get/responses/default: All of the oneOf rules match. at /tmpfs/.cpan-build-cpansand/2016080806/Swagger2-0.84-0/blib/lib/Mojolicious/Plugin/Swagger2.pm line 80, <DATA> line 565.
not ok 1 - Could not load Swagger2 plugin using YAML::XS
#   Failed test 'Could not load Swagger2 plugin using YAML::XS'
#   at t/plugin-yaml.t line 30.
# YAML::Syck 1.29
ok 2 - GET /v1/pets
ok 3 - 200 OK
ok 4 - exact match for JSON Pointer "/0/id"
ok 5 - exact match for JSON Pointer "/0/name"
1..5
# Looks like you failed 1 test of 5.
```

Then check a couple of times if it always fails with this hash seed:
```
$ env PERL_HASH_SEED=25 PERL_PERTURB_KEYS=NO perl5.24.0 -Mblib t/plugin-yaml.t
```
The hash seed seems to be different for different perl versions, it seems.

**Comment by jhthorsen** [comment]
Oh! This is awful. I'm afraid I don't have time to look at it right now, but will for sure look at it during this week.

Thanks for figuring it out 👍

**Comment by jhthorsen** [comment]
Thank you! This should be fixed in JSON::Validator 0.82.

https://github.com/jhthorsen/json-validator/commit/3469563c27d3296f7acbd1df0a293406691c460c
